lms/djangoapps/courseware/block_render.py

```diff
@@ -489,6 +489,7 @@ def inner_get_block(block):
             grade_bucket_type=grade_bucket_type,
             static_asset_path=static_asset_path,
             user_location=user_location,
+            disable_staff_debug_info=disable_staff_debug_info,
```

Comment:
This is an interesting part. Removing the call to `block.static_asset_path` changed where the field-data cache is initialized. This altered the block runtime initialization order, and the `staff_debug_info` became overridden by parent blocks. However, adding this missing argument here fixes the issue.

Commit with a temporary fix I used for the verification: https://github.com/open-craft/edx-platform/commit/655e104220d95bc24139795b549cd6af36b86f44.

Reply:
Actually, we should not pass this, as it hides staff debug buttons in other places (which is not caught by tests). See the description of the `HACK` in the PR.